### DISCUSSION TITLE
Get list of all users

### DISCUSSION THREAD

**Discussion by MarcPinnell** [codeigniter4/shield#474] [Q&A] [answered]
New to Shield, but finding my way. I can get a list of users, add new ones and edit users, but am running into a little issue with filtering the list of users. Some of the field below are fields I added and extended the UserModel.

If I do this:
```php
$users = $this->userModel	->select('users.*')
						->where('is_superadmin',0)
						->groupBy('users.id')
						->orderBy('lname,fname')
						->findAll();
```

I get SQL that looks like this:
```sql
SELECT `users`.*
FROM `users`
WHERE `is_superadmin` = 0
AND `users`.`deleted_at` IS NULL
GROUP BY `users`.`id`
ORDER BY `lname`, `fname`
```

Where is the users.deleted_at line coming from? I need to do different searches with that on and off, but can't figure out how to override it. 

TIA
Marc

**Comment by kenjis** [comment, marked as answer]
UserModel uses soft deletes. So it is necessary unless you want to get deleted users.
See https://codeigniter4.github.io/CodeIgniter4/models/model.html#usesoftdeletes

If you really need to remove it, ~~you need to extend `CodeIgniter\Model`.~~
use `withDeleted()`.

**Comment by MarcPinnell** [comment]
Perfect! Exactly what I was looking for! I have a toggle on the listing page that allows those that have been "deleted" to be shown.

**Reply by kenjis** [reply]
As you may have already noticed, there is no need to extend Model, you can use the `withDeleted()` method.

**Reply by MarcPinnell** [reply]
Yep. Actually just figured out how to do that. Both conditionally in the listing and when pulling by ID.

```
$users = $this->userModel	->select('users.*')
									->where('is_superadmin',0)
									->groupBy('users.id')
									->orderBy('lname,fname');
		if ($status == 'all') {
			$users = $this->userModel->withDeleted();
		}
		$users = $this->userModel	->findAll();
```

`$user  = $this->userModel->withDeleted()->findById($id);`

If you have any suggestions to make that more efficient or "prettier" - feel free to point it out!! Learn something new everyday about CodeIgniter! Love this! :)
